### DISCUSSION TITLE
Mulit-Select Picker: Preselect Values

### DISCUSSION THREAD

**Discussion by Bernie1502** [Budibase/budibase#3567] [Help]
Hi all,
i try Budibase to build an simple WebApp and it looks great so far :)

mySQL-Database:
table 1
table1_id, title, price, ...

table 2
table2_id, name

table 3
table1_id (FK)
table2_id (PK)

The many-to-many Relationship is set up and working.

In a Form with a data provider to table 1 i get all form fields from table 1.
The nested DataProvider fetches Data from Table2, to get the Options in my multi-select picker.

So far so good :)

My question:
Is there a way to get all options (rows) from table 2 in the Multi-select picker AND preselect the options which references with table 3?
The only thing what i found out, is to display only the options which references to table3 with the Budibase Relationsihp.

Thanks in advance for some ideas!

**Comment by mjashanks** [comment]
Hi @Bernie1502 

Unfortunately, there is no way to pre-select relationships. I have heard similar requests before though - so I'd really like to know more, if you can share it.

How do you know which ones should be pre-selected? Just a static list, or are they determined somehow? It'd be great to hear the actual use-case, (i.e. beyond `table1` & `table2`) and why this pre-selection makes sense. No problem if it's not something that you can share.

Thanks!

**Comment by Bernie1502** [comment]
Hi @mjashanks 
thanks for your reply!

In `table1` are courses, in `table2` are targetgroups. So a course can have many targetgroups and a targetgroup can belong to many courses. With the pre-selection in the multi-select picker i want to display which targetgroups are assigned to the course.
It should be also possible to (de-)assign the targetgroup to/from the course.

Sorry, with more practice i noticed, that i didn't understand the part of Relationships in BudiBase in complete...so, perhaps it is much more easier to achieve the goal :)

Thanks!

**Comment by tomwhita** [comment]
On a form in my app, I want to preselect the currently logged-in user from a relationship field (there can only be a single user selected) and disable the input so that the choice can't be modified.  Is that possible?  How?

**Reply by mjashanks** [reply]
Hi there. This is possible. First setup your component structure as follows
```
> Data Provider : data = Users
   > Options Picker - settings below
```

Options provider settings:
- Options Source = Data Provider
- Options Provider = your data provider above
- Default Value = `{{ Current User._id }}``

![image](https://user-images.githubusercontent.com/3524181/147537664-332e9dbb-cd59-43e3-a40a-ec432c73a0e2.png)
